src/lightning_app/utilities/load_app.py

```diff
@@ -113,6 +115,27 @@ def _create_fake_main_module(script_path):
     return module
 
 
+@contextmanager
+def _patch_sys_argv():
+    from lightning_app.cli.lightning_cli import run_app
+
+    original_argv = sys.argv
+    if "--app_args" not in sys.argv:
+        new_argv = [sys.executable] + sys.argv[3:4]
+    else:
```

Comment:
I gotta say, I'm not too versed in app, but this almost seems to me like you're trying to roll your own argument parsing, which is never a good idea. Can't you get parsed arguments from click context somehow? (And I mean it as a genuine question, I don't know if it's possible with the way what's being passed around and what's not.)

Reply:
I don't know if it is possible. This seemed the simplest way to get things done reliably.